File: mobile/src/main/java/com/windscribe/mobile/connectionsettings/ConnectionSettingsPresenter.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.connectionsettings;

import android.content.Context;

public interface ConnectionSettingsPresenter {

    void init();

    void onAllowLanClicked();

    void onAutoFillPacketSizeClicked();

    void onAutoStartOnBootClick();

    void onConnectionModeAutoClicked();

    void onConnectionModeManualClicked();

    void onDestroy();

    void onGpsSpoofingClick();

    void onHotStart();

    void onKeepAliveAutoModeClicked();

    void onKeepAliveManualModeClicked();

    void onManualLayoutSetupCompleted();

    void onPacketSizeAutoModeClicked();

    void onPacketSizeManualModeClicked();

    void onPermissionProvided();

    void onPortSelected(String protocol, String port);

    void onProtocolSelected(String protocol);

    void onSplitTunnelingOptionClicked();

    void onStart();

    void saveKeepAlive(String keepAlive);

    void setKeepAlive(String keepAlive);

    void setPacketSizeManual(String size);

    void setTheme(Context context);

    void onDecoyTrafficClick();

    void turnOnDecoyTraffic();

    void onFakeTrafficVolumeSelected(String label);
}

File: mobile/src/main/java/com/windscribe/mobile/connectionsettings/ConnectionSettingsPresenter.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.connectionsettings

import android.content.Context

interface ConnectionSettingsPresenter {
    fun init()
    fun onAllowLanClicked()
    fun onAutoFillPacketSizeClicked()
    fun onAutoStartOnBootClick()
    fun onConnectionModeAutoClicked()
    fun onConnectionModeManualClicked()
    fun onDestroy()
    fun onGpsSpoofingClick()
    fun onHotStart()
    fun onKeepAliveAutoModeClicked()
    fun onKeepAliveManualModeClicked()
    fun onManualLayoutSetupCompleted()
    fun onPacketSizeAutoModeClicked()
    fun onPacketSizeManualModeClicked()
    fun onPermissionProvided()
    fun onPortSelected(heading: String, port: String)
    fun onProtocolSelected(heading: String)
    fun onSplitTunnelingOptionClicked()
    fun onStart()
    fun saveKeepAlive(keepAlive: String)
    fun setKeepAlive(keepAlive: String)
    fun setPacketSize(size: String)
    fun setTheme(context: Context)
    fun onDecoyTrafficClick()
    fun turnOnDecoyTraffic()
    fun onFakeTrafficVolumeSelected(label: String)
    fun onNetworkOptionsClick()
    fun onAntiCensorshipClick()
}